src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -0,0 +1,158 @@
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.connector.ConnectorAction;
+import org.opensearch.ml.common.transport.connector.MLCreateConnectorInput;
+import org.opensearch.ml.common.transport.connector.MLCreateConnectorResponse;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import static org.opensearch.flowframework.common.CommonValue.ACTIONS_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.CREDENTIALS_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.DESCRIPTION;
+import static org.opensearch.flowframework.common.CommonValue.NAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PARAMETERS_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PROTOCOL_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.VERSION_FIELD;
+
+/**
+ * Step to create a connector for a remote model
+ */
+public class CreateConnectorStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateConnectorStep.class);
+
+    private MachineLearningNodeClient mlClient;
+
+    static final String NAME = "create_connector";
+
+    /**
+     * Instantiate this class
+     * @param mlClient client to instantiate MLClient
+     */
+    public CreateConnectorStep(MachineLearningNodeClient mlClient) {
+        this.mlClient = mlClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws IOException {
+        CompletableFuture<WorkflowData> createConnectorFuture = new CompletableFuture<>();
+
+        ActionListener<MLCreateConnectorResponse> actionListener = new ActionListener<>() {
+
+            @Override
+            public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
+                logger.info("Created connector successfully");
+                createConnectorFuture.complete(
+                    new WorkflowData(Map.ofEntries(Map.entry("connector-id", mlCreateConnectorResponse.getConnectorId())))
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to create connector");
+                createConnectorFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));
```

Comment:
Let's use [ExceptionsHelper.status](https://github.com/opensearch-project/OpenSearch/blob/1e28738b8c966011bf1ae1f00431f0377761cb0a/libs/core/src/main/java/org/opensearch/ExceptionsHelper.java#L95) to determine the RestStatus code of the exception